### PR TITLE
[Core][StructuralMechanicsApplications] Adding variables of #2063

### PR DESCRIPTION
**Pull request by loumalouomega**
Adding variables of #2063

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 1]
```diff
@@ -599,6 +599,78 @@ void BaseSolidElement::CalculateOnIntegrationPoints(
...
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, UseElementProvidedStrain());
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+
+        // Reading integration points
+        const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(  );
+
+        //Calculate Cauchy Stresses from the FE solution
+        std::vector<Vector> sigma_FE_solution(number_of_nodes);
+        Variable<Vector> variable_stress = CAUCHY_STRESS_VECTOR;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:624]: what is `CAUCHY_STRESS_VECTOR`?

File: kratos/sources/variables.cpp [hunk 2]
```diff
@@ -816,7 +821,12 @@ void KratosApplication::RegisterVariables() {
     //--------------- Meshing ApplicationApplication -------------------//
 
     KRATOS_REGISTER_VARIABLE(NODAL_ERROR)
-    KRATOS_REGISTER_VARIABLE(NODAL_ERROR_COMPONENTS)
+    KRATOS_REGISTER_3D_VARIABLE_WITH_COMPONENTS(NODAL_ERROR_COMPONENTS)
+    KRATOS_REGISTER_VARIABLE(ELEMENT_ERROR )
+    KRATOS_REGISTER_VARIABLE(ELEMENT_H )
+    KRATOS_REGISTER_VARIABLE(RECOVERED_STRESS )
+    KRATOS_REGISTER_VARIABLE(ERROR_INTEGRATION_POINT )
+    KRATOS_REGISTER_VARIABLE(CONTACT_PRESSURE )
```
Comment (philbucher) [inline kratos/sources/variables.cpp:829]: this does not exist yet?
Arent the DEM guys using sth like this?

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:624]: Is the vector containing (following Voigt notation) the Cauchy stresses

File: kratos/sources/variables.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/sources/variables.cpp:829]: imagen
Nope

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 3]
```diff
@@ -599,6 +599,78 @@ void BaseSolidElement::CalculateOnIntegrationPoints(
...
+            for (IndexType node_number=0; node_number<number_of_nodes; node_number++) {
+                const auto& sigma_recovered_node = GetGeometry()[node_number].GetValue(RECOVERED_STRESS);
+                for (IndexType stress_component = 0; stress_component<strain_size; stress_component++) {
+                    sigma_recovered[stress_component] += this_kinematic_variables.N[node_number] * sigma_recovered_node[stress_component];
+                }
+            }
+
+            // Calculate error_sigma
+            Vector error_sigma(strain_size);
+            error_sigma = sigma_recovered - sigma_FE_solution[point_number];
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:659]: Noalias?

**Review by philbucher — APPROVED** [review]
Seems ok

**Comment by loumalouomega** [comment]
Thanks, I forgot to merge

**Pull request merged**